docs/source/concept_models/vivarium_census_synthdata/concept_model.rst

```diff
@@ -479,20 +479,272 @@ When we reach that point, we might also want to think about the change
+
+        # make zero probabilities go away
+        s_name_pr = pd.Series(np.array(p), index=df_census_names.name)
+        s_name_pr = s_name_pr[s_name_pr > 0]
+        s_name_pr /= s_name_pr.sum()
+        return np.random.choice(s_name_pr.index, p=s_name_pr).capitalize()
+
+    # should everyone in a household have the same last name?  seems overly normative, but what is smarter?
+    for hh_id, dfg in df_population.groupby(['household_id']):
+        last_name = random_last_name(dfg.race_eth.value_counts().iloc[0])  # HACK: use most common race/eth in household
+        df_population.loc[dfg.index, 'last_name'] = last_name
+        # TODO: for rows with relshipp value of 22, 24, 31, 32, 34, 35, 36, give different last name
+
+**First and middle names**
+
+First names from babies:
+https://www.ssa.gov/oact/babynames/limits.html ; this page links to a
+data file of State-specific birth certificate frequencies for first
+names https://www.ssa.gov/oact/babynames/state/namesbystate.zip
+
+How to get realistic race/ethnicity for first and middle names?  And
+is that important? We could use ecological approach to back out
+race/ethnicity from state-to-state variation in first names.  To test,
+we would take (for example) a traditionally Black first name and see
+if the state-to-state rate is correlated with the percent of Black
+babies --- can use state random effects to include data from multiple
+years to be increase predictive validity.
+
+Use middle names from same distribution as first names (?). It would
+be nice to get some of the national/ethnic challenges right, like
+people from South America with many names getting their middle names
+used as different last names.
+
+We might want to eventually include nicknames and suffixes like Jr. and III.
+
+.. sourcecode:: python
+
+    # first and middle names
+    # strategy: calculate year of birth based on age, use it with sex and state to find a representative name
+    df_ssn_names = pd.read_csv('/home/j/Project/simulation_science/prl/data/ssn_names/FL.TXT',
+                               names=['state', 'sex', 'yob', 'name', 'freq'])
+    df_ssn_names['age'] = 2020 - df_ssn_names.yob
+    df_ssn_names['sex'] = df_ssn_names.sex.map({'M':1, 'F':2})
+    g_ssn_names = df_ssn_names.groupby(['age', 'sex'])
+    def random_names(age, sex, size):
+        t = g_ssn_names.get_group((age, sex))
+        p = t.freq / t.freq.sum()
+        return np.random.choice(t.name, size=size, replace=True, p=p)
+    for (age,sex), df_age in df_population.groupby(['age', 'sex']):
+        df_population.loc[df_age.index, 'first_name'] = random_names(age, sex, len(df_age))
+        df_population.loc[df_age.index, 'middle_name'] = random_names(age, sex, len(df_age))
+
+
+2.3.7 Component 11: Date of Birth
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To create a date-of-birth column in the synthetic output data, each
+simulant should have a uniformly random date of birth which is
```

Comment:
Would it be worth using an actual distribution of birthdays throughout the year? I think there are slight seasonal differences in density!